lua/neo-tree/ui/renderer.lua

```diff
@@ -1033,7 +1033,16 @@ M.acquire_window = function(state)
   end
 
   if win ~= nil then
-    vim.api.nvim_buf_set_name(state.bufnr, bufname)
+    local bufnameExists = false
+    for _, bufnr in ipairs(vim.api.nvim_list_bufs()) do
+      if vim.api.nvim_buf_is_loaded(bufnr) or vim.api.nvim_buf_get_name(bufnr) == bufname then
+        bufnameExists = true
+        break
+      end
+    end
+    if not bufnameExists then
```

Comment:
You can't just not give the buffer a name. The name is important. This will fix one bug while creating new ones.

I've been meaning to circle back on this myself but haven't found the time. See these for history f you want to spend the time:

#1418 
#1408
#1406
#1398
#1390

Reply:
Now I'm just reusing the already existed buffer name, that will fix this error:
`[Neo-tree ERROR] debounce  filesystem_navigate  error:  ...e-bug/plugins/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1036: Vim:E95: Buffer with this name already exists`
Since I'm using my own fork with these changes, I'm not getting any unexpected error or behavior. I checked almost features of Neotree in Arch Linux. I think everything is fine!